2226/2/versus.rb

```diff
@@ -0,0 +1,197 @@
+    'блядословник', 'блядский', 'блядство', 'въебаться', 'взбляд', 'впиздячил', 'выблядовал', 'выблядок',
+    'выебон', 'выёбывается', 'глупизди', 'доебался', 'ебанёшься', 'ебанул', 'ебашит', 'ёбнул', 'ебало',
+    'ебанулся', 'ебнул', 'жидоёб', 'жидоёбский', 'заёб', 'изъебнулся', 'заебал', 'заебись', 'козлоёб',
+    'козлоёбище', 'хуй', 'пизда', 'пиздец', '\*', 'сука'
+  ].freeze
+
+  SWEAR_WORDS = SWEAR_WORDS_ARRAY.join('|')
+
+  def filter_rapers_from_the_list(file_name)
+    first_filter = file_name.sub(/ротив_/, '')
+    second_filter = first_filter.sub(/_\(.*/, '')
+    third_filter = second_filter.sub(%r/^rap-battles\/_/, '')
+    third_filter.sub(/_$/, '')
+  end
+
+  def list_all_rapers
+    raper_storage = []
+
+    Dir.glob('rap-battles/*') do |file_name|
+      filtered_data = filter_rapers_from_the_list(file_name)
+      raper_storage << filtered_data.split('_п')
+      raper_storage.flatten!.uniq!
+    end
+    raper_storage
+  end
+
+  def number_of_battles(name)
+    battles_storage = [1]
+
+    Dir.glob('rap-battles/*') do |file_name|
+      battles_storage << file_name.match(name)
+      battles_storage.reject!(&:nil?)
+    end
+    battles_storage.size / 2
+  end
+
+  def number_of_swear_words(name)
+    swear_words_storage = []
+
+    Dir.glob('rap-battles/*') do |file_name|
+      if file_name.match(name)
+        text = File.read(file_name)
+        swear_words_storage << text.scan(/#{SWEAR_WORDS}/)
+        swear_words_storage.flatten!
+      end
+    end
+    swear_words_storage.size
+  end
+
+  def average_number_swearing_words_in_battle(name)
+    average = number_of_swear_words(name).fdiv(number_of_battles(name))
+    average.round(2)
+  end
+
+  def number_of_words_in_rounds(name)
+    number_of_words_in_rounds = 0
+
+    Dir.glob('rap-battles/*') do |file_name|
+      if file_name.match(name)
+        text = File.read(file_name)
+        number_of_words_in_rounds += text.split(' ').length
+      end
+    end
+    number_of_words_in_rounds
+  end
+
+  def number_of_rounds(name)
+    round_words = []
+
+    Dir.glob('rap-battles/*') do |file_name|
+      if file_name.match(name)
+        text = File.read(file_name)
+        round_words << text.match(/Раунд/)
+        round_words.reject!(&:nil?)
+      end
+    end
+    round_words
+  end
+
+  def average_number_words_in_round(name)
+    number_of_rounds = number_of_rounds(name).size
+    number_of_rounds = 1 if number_of_rounds.zero?
+
+    number_of_words_in_rounds(name) / number_of_rounds
+  end
+
+  def sort_and_reverse_hash(hash_for_sorting, sorted_storage)
+    hash_for_sorting = hash_for_sorting.sort_by { |_key, value| value[1] }
+    hash_for_sorting.reverse!
+    hash_for_sorting.each do |elem|
+      sorted_storage[elem[0]] = elem[1]
+    end
+    sorted_storage
+  end
+
+  def statistics(number)
+    unsorted_storage = {}
+    sorted_storage = {}
+    list_all_rapers.each do |raper|
```

Comment:
each_with_object
and why not just `unsorted_storage[raper] << [number_of_battles(raper), number_of_swear_words(raper), ...]`